pinot-core/src/main/java/com/linkedin/pinot/core/common/BlockValSet.java

```diff
@@ -38,7 +60,26 @@
    * @param outStartPos starting index position in outDictionaryIds. Indexes will
    *                    be copied starting at this position.
    *                    outDictionaryIds must be atleast (outStartPos + inDocIdsSize) in size
+   * TODO: Remove arguments from this api, as ProjectionBlock has all the required info.
    */
-  void readIntValues(int[] inDocIds, int inStartPos, int inDocIdsSize, int[] outDictionaryIds, int outStartPos);
+  void getDictionaryIds(int[] inDocIds, int inStartPos, int inDocIdsSize, int[] outDictionaryIds, int outStartPos);
```

Comment:
Good to have a saperate API for getting dictionary IDs. Make code much more readable :+1: